go/pkg/rexec/rexec.go

```diff
@@ -397,16 +397,24 @@ func (ec *Context) ExecuteRemotely() {
 		ec.setOutputMetadata()
 		ec.Result = command.NewResultFromExitCode((int)(ec.resPb.ExitCode))
 		if ec.opt.DownloadOutErr {
-			streamOut.Do(func() {
-				if err := ec.downloadStream(ec.resPb.StdoutRaw, ec.resPb.StdoutDigest, ec.oe.WriteOut); err != nil {
-					ec.Result = command.NewRemoteErrorResult(err)
-				}
-			})
-			streamErr.Do(func() {
-				if err := ec.downloadStream(ec.resPb.StderrRaw, ec.resPb.StderrDigest, ec.oe.WriteErr); err != nil {
-					ec.Result = command.NewRemoteErrorResult(err)
-				}
-			})
+			offset := int(nOutStreamed)
+			if int64(offset) != nOutStreamed {
+				// Offset is larger than math.MaxInt, the stream cannot be read anyways.
+				// Resetting to 0 here to allow the error from downstream to bubble back up here.
+				offset = 0
+			}
+			if err := ec.downloadStream(ec.resPb.StdoutRaw, ec.resPb.StdoutDigest, offset, ec.oe.WriteOut); err != nil {
```

Comment:
Should we check whether streaming already succeeded -- perhaps if offset == digest.SizeBytes -- and not download if it has?

Reply:
There is not such optimization downstream, which is not what I expected. I suppose it's good to add one here.